Update MpTelemetry JaegerQueryClient to retry before throwing exception

Currently 
[runWithRetryAndTimeout](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java#L308) throws an exception on the first try. Defect 303752 shows this error `Caused by: io.netty.handler.codec.http2.Http2Exception: Error decoding headers: decode only works with an entire header block!` intermittently so hopefully retrying will make the test a bit more reliable.

--- a/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2022 IBM Corporation and others.
+ * Copyright (c) 2022, 2025 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -310,7 +310,10 @@ public class JaegerQueryClient implements AutoCloseable {
         while (retryCount < 5) {
             Log.info(c, "runWithRetryandTimeout", "retryCount " + retryCount);
             retryCount += 1;
-	@@ -318,8 +317,11 @@ private static <T> T runWithRetryAndTimeout(Callable<T> action, Duration timeout
+            //get executOR
+            ExecutorService e = Executors.newSingleThreadExecutor();
+            Future<T> future = e.submit(action);
+            //
             try {
                 return future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
             } catch (InterruptedException | ExecutionException e1) {

--- a/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
@@ -305,29 +305,25 @@ public class JaegerQueryClient implements AutoCloseable {
      * @return the result of the action
      * @throws RuntimeException if the action throws an exception or times out five times
      */
-    private static <T> T runWithRetryAndTimeout(Callable<T> action, Duration timeout) {
+    private static <T> T runWithRetryAndTimeout(Callable<T> action, Duration timeout){
         int retryCount = 0;
-
         while (retryCount < 5) {
             Log.info(c, "runWithRetryandTimeout", "retryCount " + retryCount);
             retryCount += 1;
-            //get executOR
-            ExecutorService e = Executors.newSingleThreadExecutor();
-            Future<T> future = e.submit(action);
-            //
+	@@ -318,8 +317,11 @@ private static <T> T runWithRetryAndTimeout(Callable<T> action, Duration timeout
             try {
                 return future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
             } catch (InterruptedException | ExecutionException e1) {
+                future.cancel(true);//interrupt the task if it's still running
+                Log.warning(c,"runWithRetryandTimeout: Action threw an exception " + e1);
                 if(retryCount == 5){
-                    future.cancel(true); //interrupt the task if it's still running
                     throw new RuntimeException("Action threw an exception ", e1);
-                }
+                }         
             } catch (TimeoutException e1) {
                 future.cancel(true);
             } finally {
                 e.shutdown();
             }
-
         }
         throw new RuntimeException("Action timeout");
     }

--- a/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java
@@ -295,7 +295,7 @@ public class JaegerQueryClient implements AutoCloseable {
     }
 
     /**
-     * Run an action asynchronously with a timeout, retrying up to three times if it times out.
+     * Run an action asynchronously with a timeout, retrying up to five times if it times out.
      * <p>
      * Useful for methods which use the raw client to talk to Jaeger
      *
@@ -303,7 +303,7 @@ public class JaegerQueryClient implements AutoCloseable {
      * @param action the action to run
      * @param timeout the timeout for each retry
      * @return the result of the action
-     * @throws RuntimeException if the action throws an exception or times out three times
+     * @throws RuntimeException if the action throws an exception or times out five times
      */
     private static <T> T runWithRetryAndTimeout(Callable<T> action, Duration timeout) {
         int retryCount = 0;
@@ -318,8 +318,10 @@ public class JaegerQueryClient implements AutoCloseable {
             try {
                 return future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
             } catch (InterruptedException | ExecutionException e1) {
-                future.cancel(true); //interrupt the task if it's still running
-                throw new RuntimeException("Action throw an exception ", e1);
+                if(retryCount == 5){
+                    future.cancel(true); //interrupt the task if it's still running
+                    throw new RuntimeException("Action threw an exception ", e1);
+                }
             } catch (TimeoutException e1) {
                 future.cancel(true);
             } finally {

